lib/charms/layer/jenkins/api.py

```diff
@@ -34,13 +34,21 @@
 class Api(object):
     """Encapsulate operations on the Jenkins master."""
 
+    @property
+    def url(self):
+        config = hookenv.config()
+        prefix = urlparse(config["public_url"]).path
+        if not prefix:
+            prefix = "/"
+        return "http://localhost:8080{}".format(prefix)
```

Comment:
Please take advantage of the URL constants defined in charms.layer.jenkins.service and use urljoin:

```python
return urljoin(URL, perfix)
```